cypress/integration/data-browser/graphs.spec.js

```diff
@@ -1,171 +1,134 @@
+    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`)
+    // Regex check graphName, periodLow, periodHigh and visibleSeries in the url
+    cy.url().then((url) => {
+      url.match(/^([a-z]{2,}-?)+$/)
+    })
+    cy.url().then((url) => {
+      url.match(/periodLow=^20\d{2}-Q\d/)
+    })
+    cy.url().then((url) => {
+      url.match(/periodHigh=^20\d{2}-Q\d/)
+    })
+    cy.url().then((url) => {
+      url.match(/visibleSeries^([a-zA-Z-\/]+(%20)?[a-zA-Z-]*,?)+$/)
+    })
+  })
+
+  it("Visit graph page then selects a graph from the dropdown menu", () => {
+    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`)
+
+    cy.get(".react-select__graph__value-container").click(0, 0, {
+      force: true,
+    })
+    cy.get("#react-select-2-option-0-0").click(0, 0, { force: true })
+    cy.url().should("include", "ltv")
+    cy.url().then((url) => {
+      url.match(/^([a-z]{2,}-?)+$/)
+    })
+  })
+
+  it("Visits graphs page, change periods, checks that the url updates with selected periods, clicks 'Show All Quarters' button and checks url updated with default periods", () => {
+    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`)
+
+    cy.url().then((url) => {
+      cy.get(".react-select__period_start__value-container").click(0, 0, {
+        force: true,
+      })
+      cy.get("#react-select-3-option-3").click(0, 0, { force: true })
+      // Check url doesn't include base graph url start quarter
+      !url.includes(url)
+
+      cy.get(".react-select__period_end__value-container").click(0, 0, {
+        force: true,
+      })
+      cy.get("#react-select-4-option-6").click(0, 0, { force: true })
+      // Check URL doesn't include base graph url end quarter
+      !url.includes(url)
+
+      cy.get(".reset").click(0, 0, { force: true })
+      // Checks url was reset to base graph url which includes start and end quarters
+      url.includes(url)
+    })
+  })
+
+  it("De-select and re-select a series, UI and URL updates", () => {
+    // let urlUpdate
+    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`)
+    // De-select 'Conventional Conforming' from series
+    cy.get(".highcharts-color-0 > text").click(0, 0, { force: true })
+    cy.url().should("not.contain", "Conventional%20Conforming")
+    // Re-select 'Conventional Conforming' from series
+    cy.get(".highcharts-color-0 > text").click(0, 0, { force: true })
+    cy.url().should("include", "Conventional%20Conforming")
+  })
 
   it("URL loads with specific set of series and it's reflected in the UI", () => {
-    let series = ["FHA", "HELOC", "VA"];
     cy.visit(
       `${baseURLToVisit}/data-browser/graphs/quarterly/applications?periodLow=2018-Q1&periodHigh=2021-Q4&visibleSeries=FHA,HELOC,VA`
-    );
-    cy.wait(1000);
-    cy.url("include", series);
-  });
-});
+    )
+    cy.url().then((url) => {
+      url.match(/visibleSeries^([a-zA-Z-\/]+(%20)?[a-zA-Z-]*,?)+$/)
```

Comment:
Never fails, even if we change `visibleSeries` to `blahrg`.

https://docs.cypress.io/api/commands/should#Syntax